cli/src/commands/new.rs

```diff
@@ -166,10 +157,22 @@ Please use `jj new 'all:x|y'` instead of `jj new --allow-large-revsets x y`.",
             .try_collect()?;
         workspace_command.check_rewritable(children_commits.iter().ids())?;
     } else {
-        parent_commits = target_commits;
+        parent_commits = workspace_command
+            .resolve_some_revsets_default_single(&args.revisions)?
+            .into_iter()
+            .collect_vec();
         parent_commit_ids = parent_commits.iter().ids().cloned().collect();
         children_commits = vec![];
         workspace_command.check_rewritable([])?;
+
+        let should_advance_branches = parent_commits.len() == 1
+            && args.insert_before.is_empty()
+            && args.insert_after.is_empty();
```

Comment:
nit: maybe you forgot to remove `insert_before`/`after.is_empty()`?

Reply:
Updated!